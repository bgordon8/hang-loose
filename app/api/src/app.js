import express from 'express'
import users from './routes/users'

const app = express()

app.use(users)

if (process.env.NODE_ENV !== 'test') {
  app.listen(3000, () => {
    console.log('listening on port 3000...')
  })
}

export default app
