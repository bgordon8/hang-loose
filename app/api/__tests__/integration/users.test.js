import supertest from 'supertest'
import app from '../../src/app'

const req = supertest(app)

describe('routes : users', () => {
  describe('GET /users', () => {
    it('returns all users', async () => {
      const res = await req.get('/users')

      expect(res.status).toBe(200)
      expect(res.body).toHaveProperty('status')
      expect(res.body.status).toBe('success')
      expect(res.body).toHaveProperty('users')
      expect(res.body.users[0]).toHaveProperty('id')
      expect(res.body.users[0].id).toBe(1)
      expect(res.body.users[0]).toHaveProperty('email')
      expect(res.body.users[0].email).toBe('leo@email.com')
      expect(res.body.users[0]).toHaveProperty('username')
      expect(res.body.users[0].username).toBe('Leonardo')
      expect(res.body).toMatchSnapshot()
    })
  })
})
