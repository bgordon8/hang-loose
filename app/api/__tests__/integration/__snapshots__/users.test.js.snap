// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes : users GET /users returns all users 1`] = `
Object {
  "status": "success",
  "users": Array [
    Object {
      "created_at": "1984-07-15T06:00:00.000Z",
      "email": "leo@email.com",
      "id": 1,
      "password": "password123",
      "updated_at": "1984-07-15T06:00:00.000Z",
      "username": "Leonardo",
    },
    Object {
      "created_at": "1984-07-15T06:00:00.000Z",
      "email": "don@email.com",
      "id": 2,
      "password": "password123",
      "updated_at": "1984-07-15T06:00:00.000Z",
      "username": "Donatello",
    },
    Object {
      "created_at": "1984-07-15T06:00:00.000Z",
      "email": "mikey@email.com",
      "id": 3,
      "password": "password123",
      "updated_at": "1984-07-15T06:00:00.000Z",
      "username": "Michelangelo",
    },
    Object {
      "created_at": "1984-07-15T06:00:00.000Z",
      "email": "ralph@email.com",
      "id": 4,
      "password": "password123",
      "updated_at": "1984-07-15T06:00:00.000Z",
      "username": "Raphael",
    },
  ],
}
`;
